src/test/java/org/springframework/boot/autoconfigure/jms/hornetq/ClusteredHornetQAutoConfiguration1Test.java

```diff
@@ -25,54 +29,48 @@ public void shouldUseAutoconfigurationToCreateMissingBean() {
                 "jmsConnectionFactory");
         assertNotNull(jmsConnectionFactory);
 
-        HornetQConnectionFactory viaType = ctx.getBean(HornetQConnectionFactory.class);
+        ConnectionFactory viaType = ctx.getBean(ConnectionFactory.class);
         assertSame(jmsConnectionFactory, viaType);
     }
 
-    static class UserProvidedFactory {
-
-        static final HornetQConnectionFactory dummyFactory = mock(HornetQConnectionFactory.class);
-
-        @Bean
-        ConnectionFactory jmsConnectionFactory() {
-            return dummyFactory;
-        }
-    }
-
     @Test
-    public void shouldUseUserProvidedConfigOverAutoconfiguration() {
+    public void shouldUseAutoconfigurationToCreateMissingBeanWithCredentials() {
         ctx = new AnnotationConfigApplicationContext();
-        ctx.register(UserProvidedFactory.class, ClusteredHornetQAutoConfiguration.class);
+        Properties properties = new Properties();
+        properties.setProperty("spring.hornetq.user", "user");
+        ctx.getEnvironment().getPropertySources().addFirst(new PropertiesPropertySource(
+                "credentials", properties));
+        ctx.register(ClusteredHornetQAutoConfiguration.class);
         ctx.refresh();
-        HornetQConnectionFactory jmsConnectionFactory = (HornetQConnectionFactory) ctx.getBean(
-                "jmsConnectionFactory");
-        assertSame(UserProvidedFactory.dummyFactory, jmsConnectionFactory);
+        UserCredentialsConnectionFactoryAdapter jmsConnectionFactory = (UserCredentialsConnectionFactoryAdapter) ctx
+                .getBean("jmsConnectionFactory");
+        assertNotNull(jmsConnectionFactory);
 
-        HornetQConnectionFactory viaType = ctx.getBean(HornetQConnectionFactory.class);
-        assertSame(UserProvidedFactory.dummyFactory, viaType);
+        ConnectionFactory viaType = ctx.getBean(ConnectionFactory.class);
+        assertSame(jmsConnectionFactory, viaType);
     }
 
-    static class UserProvidedFactory2 {
+    static class UserProvidedFactory {
 
         static final HornetQConnectionFactory dummyFactory = mock(HornetQConnectionFactory.class);
 
         @Bean
-        HornetQConnectionFactory jmsConnectionFactory() {
```

Comment:
The difference in the duplicated tests has  that it bind HornetQConnectionFactory instead of ConnectionFactory.

It is OK for me to left only one test, as the added value  is probably not too big. If we re-add the test, we should make it a bit more obvious on  what the difference  between the tests are.